pkgs/development/ada-modules/gnatcoll/bindings.nix

```diff
@@ -42,6 +43,24 @@ stdenv.mkDerivation rec {
     sha256 = "00aakpmr67r72l1h3jpkaw83p1a2mjjvfk635yy5c1nss3ji1qjm";
   };
 
+  patches = [
+    (fetchpatch {
+      # Add minimal support for Python 3.11
+      url = "https://github.com/AdaCore/gnatcoll-bindings/commit/cd650de5.patch";
```

Comment:
We need to use full commit hashes here since these URLs could become ambiguous later. Similarly, GitHub patches mention commit hashes that are dynamically sized based on repository contents and unfortunately even `fetchpatch2` fails to strip them out, so please also append `?full_index=1` to get it to stop abbreviating them. The hashes will need updating for these changes.

Reply:
I got them from jj (which was tremendously useful to check patch ordering and conflicting, as expected). I'll switch to the full hashes and do the other changes when I get back to a keyboard. Thanks for the exhaustive review @emilazy.